### PR TITLE
Auth with configuration props

### PR DESCRIPTION
**Pull request by stevengibbszions**
Use configuration properties as source for instance credentials.  This will allow a more secure way to provide credentials when using spring cloud kubernetes discovery.

### CODE REVIEW

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java [hunk 1]
```diff
@@ -127,6 +129,56 @@ public String path(String path) {
...
+		 */
+		private boolean enabled = false;
+
+		/**
+		 * Default username used for authentication to each instance. Individual values
+		 * for specific instances can be overriden using
+		 * `spring.boot.admin.instance-auth.service-map.*.user-name`. <br/>
+		 * Default: client
+		 */
+		private String defaultUserName = "client";
```
Comment (joshiste) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java:148]: Imho the default value must be `null`. We should not provide default usernames/passwords

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java [hunk 2]
```diff
@@ -127,6 +129,56 @@ public String path(String path) {
...
+
+		/**
+		 * Default userpassword used for authentication to each instance. Individual
+		 * values for specific instances can be overriden using
+		 * `spring.boot.admin.instance-auth.service-map.*.user-password`. <br/>
+		 * Default: client
+		 */
+		private String defaultPassword = "client";
```
Comment (joshiste) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java:156]: Imho the default value must be `null`. We should not provide default usernames/passwords

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/web/client/BasicAuthHttpHeaderProvider.java [hunk 3]
```diff
@@ -25,6 +25,7 @@
 import org.springframework.util.Base64Utils;
 import org.springframework.util.StringUtils;
 
+import de.codecentric.boot.admin.server.config.AdminServerProperties.InstanceCredentials;
```
Comment (joshiste) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/web/client/BasicAuthHttpHeaderProvider.java:28]: I'd suggest to move the imported class to the `BasicAuthHttpHeaderProvider` class

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java [hunk 4]
```diff
@@ -127,6 +129,56 @@ public String path(String path) {
...
 
+	@lombok.Data
+	public static class InstanceAuthProperties {
+
+		/**
+		 * Whether or not to use configuration properties as a source for instance
+		 * credentials <br/>
+		 * Default: false
+		 */
+		private boolean enabled = false;
```
Comment (joshiste) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java:140]: If we don't provide any default passwords, I'd suggest to enable the feature by default.

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java (hunk 1 above)
Comment (stevengibbszions) [reply spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java]: Set to null

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java (hunk 2 above)
Comment (stevengibbszions) [reply spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java]: Set to null

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/web/client/BasicAuthHttpHeaderProvider.java (hunk 3 above)
Comment (stevengibbszions) [reply spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/web/client/BasicAuthHttpHeaderProvider.java]: Moved class

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java (hunk 4 above)
Comment (stevengibbszions) [reply spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerProperties.java]: now enabled by default, and updated the docs to reflect that.

**Pull request merged**